src/Point/Point.cs

```
@@ -156,35 +147,53 @@ public async Task PutInEndPosition()
                     {
-                        GenericDegradedPointPosition.DegradedRight => PointPosition.Right,
-                        GenericDegradedPointPosition.DegradedLeft => PointPosition.Left,
+                        GenericDegradedPointPosition.DegradedRight => GenericPointPosition.Right,
+                        GenericDegradedPointPosition.DegradedLeft => GenericPointPosition.Left,
                         GenericDegradedPointPosition.NotDegraded => null,
                         GenericDegradedPointPosition.NotApplicable => null,
                         _ => null,
                     };
                     if (finalPosition != null)
                     {
-                        UpdatePointState((GenericPointPosition)finalPosition, reportedDegradedPointPosition);
+                        _logger.LogInformation("Putting point into end position {}.", finalPosition);
+                        SetPointState(finalPosition.Value, reportedDegradedPointPosition);
                         await _connection.SendPointPosition(PointState);
                     }
                 }
             }
             await UpdateConnectedWebClients();
         }
 
+        /// <summary>
+        /// Checks if AllPointMachinesCrucial is set and returns NotApplicable if it is.
+        /// </summary>
+        /// <param name="degradedPointPosition"></param>
+        /// <returns></returns>
+        private GenericDegradedPointPosition RespectAllPointMachinesCrucial(GenericDegradedPointPosition degradedPointPosition) 
+        {
+            if (AllPointMachinesCrucial)
+            {
+                if (degradedPointPosition != GenericDegradedPointPosition.NotDegraded && degradedPointPosition != GenericDegradedPointPosition.NotApplicable) 
+                    _logger.LogWarning("All point machines are crucial, ignoring degraded point position {}.", degradedPointPosition);

```

should we throw an exception here?